src/node_file.cc

```diff
@@ -201,8 +291,19 @@ static void After(uv_fs_t *req) {
         break;
 
       case UV_FS_READLINK:
-        argv[1] = String::NewFromUtf8(env->isolate(),
-                                      static_cast<const char*>(req->ptr));
+        link = encodeFilename(env,
+                              static_cast<const char*>(req->ptr),
+                              req_wrap->encoding_);
+        if (link->IsNull()) {
+          argv[0] = UVException(env->isolate(),
+                                UV__ECHARSET,
```

Comment:
Is there a different error code you'd recommend?
On Mar 14, 2016 11:01 AM, "Ben Noordhuis" <notifications@github.com> wrote:

> In src/node_file.cc
> <https://github.com/nodejs/node/pull/5616#discussion_r56046516>:
>
> > @@ -201,8 +291,19 @@ static void After(uv_fs_t *req) {
> >          break;
> >
> >        case UV_FS_READLINK:
> > -        argv[1] = String::NewFromUtf8(env->isolate(),
> > -                                      static_cast<const char*>(req->ptr));
> > +        link = encodeFilename(env,
> > +                              static_cast<const char*>(req->ptr),
> > +                              req_wrap->encoding_);
> > +        if (link->IsNull()) {
> > +          argv[0] = UVException(env->isolate(),
> > +                                UV__ECHARSET,
>
> Don't use UV__ECHARSET.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/nodejs/node/pull/5616/files#r56046516>.
>